#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: fix monitor mode causing emacs as $EDITOR to hang on kernel 2.6
> http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=224543
> http://groups.google.com/groups?selm=KM6D.88S.3%40gated-at.bofh.it
> 
> it is said that this is a bash bug. Is there a patch for the 2.05b
> version?

Try this:

*** ../bash-2.05b-patched/jobs.c	Thu May  9 11:56:20 2002
--- ./jobs.c	Fri Dec 19 15:44:41 2003
***************
*** 1334,1353 ****
  	  if (setpgid (mypid, pipeline_pgrp) < 0)
  	    sys_error ("child setpgid (%ld to %ld)", (long)mypid, (long)pipeline_pgrp);
! #if defined (PGRP_PIPE)
! 	  if (pipeline_pgrp == mypid)
! 	    {
! #endif
! 	      /* By convention (and assumption above), if
! 		 pipeline_pgrp == shell_pgrp, we are making a child for
! 		 command substitution.
! 		 In this case, we don't want to give the terminal to the
! 		 shell's process group (we could be in the middle of a
! 		 pipeline, for example). */
! 	      if (async_p == 0 && pipeline_pgrp != shell_pgrp)
! 		give_terminal_to (pipeline_pgrp, 0);
  
  #if defined (PGRP_PIPE)
! 	      pipe_read (pgrp_pipe);
! 	    }
  #endif
  	}
--- 1359,1375 ----
  	  if (setpgid (mypid, pipeline_pgrp) < 0)
  	    sys_error ("child setpgid (%ld to %ld)", (long)mypid, (long)pipeline_pgrp);
! 
! 	  /* By convention (and assumption above), if
! 	     pipeline_pgrp == shell_pgrp, we are making a child for
! 	     command substitution.
! 	     In this case, we don't want to give the terminal to the
! 	     shell's process group (we could be in the middle of a
! 	     pipeline, for example). */
! 	  if (async_p == 0 && pipeline_pgrp != shell_pgrp)
! 	    give_terminal_to (pipeline_pgrp, 0);
  
  #if defined (PGRP_PIPE)
! 	  if (pipeline_pgrp == mypid)
! 	    pipe_read (pgrp_pipe);
  #endif
  	}

-- 
``The lyf so short, the craft so long to lerne.'' - Chaucer
( ``Discere est Dolere'' -- chet )
						Live...Laugh...Love
Chet Ramey, ITS, CWRU    chet@po.cwru.edu    http://tiswww.tis.cwru.edu/~chet/
